ntbs-service/wwwroot/source/Components/AutocompleteSelect.ts

```diff
@@ -0,0 +1,65 @@
+import Vue from "vue";
+var accessible = require("accessible-autocomplete");
+
+const AutocompleteSelect = Vue.extend({
+    props: ["placeholder", "validate"],
+    mounted: function() {
+        var selectElement = this.$el;
+        var selectId = selectElement.id;
+        accessible.enhanceSelectElement({
+            placeholder: this.$props.placeholder,
+            autoselect: false,
+            preserveNullOptions: true,
+            showAllValues: true,
+            selectElement: selectElement,
+            onConfirm: (val: any) => {
+                this.onConfirm(val);
+            }
+        })
+        this.autocompleteElement = document.getElementById(selectId) as HTMLInputElement;
+        this.selectElement = document.getElementById(this.autocompleteElement.id + "-select") as HTMLSelectElement;
+        // Workaround for stopping Chrome trying to autocomplete by country, see discussion https://gist.github.com/niksumeiko/360164708c3b326bd1c8
+        // Note that autocomplete="off" does not work.
+        this.autocompleteElement.setAttribute('autocomplete', 'no');
+        // As we do not build the element ourselves, use ordinary event listeners rather than vue's
+        if (this.$props.validate) {
+            this.autocompleteElement.addEventListener('change', (event: any) => this.inputChanged(event));
+            this.selectElement.addEventListener('select-changed', (event: any) => this.selectChanged(event));
+        }
+    },
+    methods: {
+        inputChanged: function (event: FocusEvent) {
+            if (this.autocompleteElement.value === "")
+            {
+                // Autocomplete does not automatically select blank value when deleting text rather than clicking on blank, so emit this manually
+                this.selectElement.value = "";
+                this.$emit("validate-input", event);
+            }
+        },
+        onConfirm: function (value: any) {
+            // this is called twice, value is undefined when user clicks away from input field which is when we want validation to happen
+            if (!this.$props.validate || value != undefined) {
+                return;
+            }
+            var event = new Event('select-changed');
+            var autocompleteValue = this.autocompleteElement.value;
+            // the actual select element does not have its value set at this point; the only way we seem to be able to find it is by searching for the matching autocomplete
+            // value by name in all the options
+            for (var i=0; i < this.selectElement.options.length; i++) {
+                if (autocompleteValue === this.selectElement[i].textContent) {
+                    this.selectElement.value = this.selectElement[i].value;
+                    this.selectElement.dispatchEvent(event);

```

Comment:
Is the dispatchEvent necessary to get some additional context from the DOM in the event? Or is it possible to just do an internal call to `selectChanged`?

Reply:
As mentioned above, I couldn't get any of the inbuilt listeners on the select element to fire, so I'm firing one manually

Reply:
But then it's captured by this component - I was asking if there was anything added to the 'event' object by dispatching it from the DOM element, or whether it would be possible to just call the component method directly?

Reply:
I need to dispatch an event happening on the select element, so it can be picked up by the containing validate-input vue component. The dispatchEvent line is necessary, as that's what links the event to the select field.
Sorry this explanation isn't very clear ha, but validate input doesn't have access to the select field if I just call this.$emit("validate-input", event) here without calling dispatchEvent first